articles/node-vs-python/index.md

```diff
@@ -0,0 +1,179 @@
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/node-vs-python/hero.jpg
+    alt: node.js vs Python
+---
+JavaScript and Python are the most popular languages among web developers. The idea of separating the front-end from the back-end as much as possible has birthed the idea of the JAMStack. JAMStack stands for **JavaScript, APIs and Markup**. It describes the idea of a website being completely static and client-side rendered. The website uses API calls to load all dynamic content. Let us look at the two most popular languages for writing APIs, Node and Python to compare and contrast the two.
+<!--more-->
+
+### Table of Contents
+
+- [The History](#the-history)
+- [Node vs Python: The Differences](#node-vs-python:-the-differences)
+- [APIs in Node](#apis-in-node)
+- [APIs in Python](#apis-in-python)
+- [Further Reading](#further-reading)
+
+### The History
+
+#### Node.js
+Node.js is a server-side JavaScript runtime. JavaScript was originally designed to run within a browser, inside a JavaScript engine like V8. To expand the capabilities of JavaScript outside the browser, Node.js was developed in **2009** by **Ryan Dahl**. Node.js combined the capabilities of the V8 JavaScript engine, an event loop and low level calls into a robust server-side scripting runtime. It soon grew into the preferred language by web developers.
+
+#### Python
+Python is a multi-paradigm, scripting language conceived by **Guido van Rossum** in the late 1980s. Python is extremely easy to read, straightforward and simple to learn. It was built with simplicity in mind. It thus uses **indentation** to define scope of loops and functions. Python is extensively used as a server-side language. Frameworks like **Django** and **Flask** make it simple to design performant websites. It is also used to create APIs and Machine Learning models. Python, combined with the package manager **pip** make it perfect for projects.
+
+### Node.js vs Python: The Differences
+- [Performance and Architecture](#performance-and-architecture)
+- [Syntax and Ease of Use](#syntax-and-ease-of-use)
+- [Type of Projects](#type-of-projects)
+- [Frameworks and Expandability](#frameworks-and-expandability)
+- [Package Management](#package-management)
+
+#### Performance and Architecture
+Python and Node.js are interpreted languages. Therefore, they are comparatively slower than compiled languages like C++.
+
+Python is a single threaded and single flowing language. Though it is optimized by frameworks such as Django, it is slow. Python runs well on servers, but is quite unsuitable for mobile. The limit in processing power is a bottleneck to Python's performance.
+
+Node.js has the upper hand here. Node.js is heavily optimized by Google's V8 JavaScript Engine, making it performant. Node.js follows an event loop based architecture, which makes it non-blocking. Asynchronous requests can be handled. This means that the runtime can execute multiple requests without getting blocked.
```

Comment:
"Asynchronous requests are supported and this means..." is a little bit more correct (and sounds better IMO).

Reply:
Sure, shall make the suggested change! 👍